src/main/java/org/folio/circulation/domain/CheckInContext.java

```diff
@@ -237,4 +241,23 @@ public ItemStatus getItemStatusBeforeCheckIn() {
   public boolean areLostItemFeesRefundedOrCancelled() {
     return lostItemFeesRefundedOrCancelled;
   }
+
+  public JsonObject asJson() {
+    JsonObject json = new JsonObject();
+    write(json, "item", JsonObject.mapFrom(item));
+    write(json, "loan", loan.asJsonWithUserAndProxy());
+    write(json, "checkInRequest", JsonObject.mapFrom(checkInRequest));
+    write(json, "requestQueue", JsonObject.mapFrom(requestQueue));
+    write(json, "checkInServicePoint", JsonObject.mapFrom(checkInServicePoint));
```

Comment:
I think it won't serialize the object properly because the object holds raw json representation.